### ISSUE TITLE
[L] A&R: Add Date Range search/filter to export data page

### ISSUE THREAD

**Issue by JLWade** [bcgov/bcparks-ar-admin#254]
#### Description:
As a bc parks staff who uses A&R, I would like to export a specific date range in a standard report so that I do not download all the data from A&R.

This ticket is to add a date range filter (from admin library) to the export report standard report tab.

downloading an A&R standard report is getting big, especially when users do not need to see all data from all years. 

#### Acceptance Criteria:

Given I am a A&R admin user (role dependent for what data is downloaded)
When I go to the export report page
Then I see an option to select a date range OR export all data on the standard report tab.

Given I am an A&R admin user (role dependent for what data is downloaded)
When I go to the export report page and select date range for standard report download
Then I can select a specific date range by year and month to download.

#### Development Checklist:

- [ ] ...
- [ ] ...
- [ ] ...

#### Dependencies

- Blocked by https://github.com/bcgov/bcparks-ar-api/issues/326
- Blocking

**Wireframe(s):**
[Figma Design](https://www.figma.com/design/yoMhvO0KZvsM6qoqwuIjpt/Attendance-%26-Revenue?node-id=6098-48961&t=K9ox8LrekrTsbUsf-1)
- the filter should only be available if Export All is unchecked. 
- Prototype for dropdown behaviour is unavailable

**Workflows:**

**Definition of Ready** (Note: If any of these points are not applicable, mark N/A)

- [ ] User story is included
- [ ] User role and type are identified
- [ ] Acceptance criteria are included
- [ ] Wireframes are included
- [ ] Design / Solution is accepted by Product Owner
- [ ] Dependencies are identified (technical, business, regulatory/policy)
- [ ] Story has been estimated (under 13 pts)

**Definition of Done** 
- In progress:
  - [ ] Acceptance criteria are tested (Functionality meets the acceptance criteria defined in the ticket)
  - [ ] UI meets accessibility requirements
  - [ ] Unit tests are written
  - [ ] Work is traceable in GitHub
  - [ ] PR linked to ticket number
  - [ ] If needed/required - Dev adds flag/label to highlight any migration steps necessary prior to PROD deployment
- Code review:
  - [ ] Code is peer reviewed and has passed CI/CD tests
- QA:
  - [ ] Acceptance criteria are tested (Functionality meets the acceptance criteria defined in the ticket)
  - [ ] Code is potentially shippable to the production environment
  - [ ] Functional features have been tested and passed by QA
  - [ ] UI components tested by designer
  - [ ] Code is deployed to PROD when moved to 'done' column (unless requested otherwise by PO)
- PO Review:
  - [ ] Acceptance criteria are tested (Functionality meets the acceptance criteria defined in the ticket)
  - [ ] Reviewed and approved by Product Owner

#### Notes:
- 
-

**Comment by Dianadec** [comment]
Q for team in refinement: Does something like this need a separate design ticket, or is it straight forward enough to embed the full implementation in this one card? cc @JLWade

**Comment by Dianadec** [comment]
Refinement Notes:
- 99% of the time the export is just for the most recent data.
- Prefers to export a set of months (eg: Jan-March as an example of a set)
- Rarely would they need to export the whole thing

**Comment by meyerdarcie** [comment]
Designs in ticket #338

**Comment by Dianadec** [comment]
Blocked by SAM release.

**Comment by Dianadec** [comment]
This may have been a duplicate of this one - https://github.com/bcgov/bcparks-ar-admin/issues/265

**Comment by Dianadec** [comment]
@meredithom seeing this has a couple PRs, should this ticket be in the Code Review pipeline?

**Comment by manuji** [comment]
Tested on [TEST](https://test-ar.bcparks.ca/export-reports): **Failed**

* Verified that the date range field only enables when the checkbox to export all is unchecked. 

**Issues:**

* Exporting a standard report with "All" or a date range export empty reports, tested with both sysAdmin user and a Park/facility specific role. 
* The font size of the label "Export all" looks smaller than whats given on Figma. 
* When the date range have lengthy month names, the text overflow out of the field. 
* The date picket shows the month name "September" in two lines. 

![Image](https://github.com/user-attachments/assets/5c49b206-0ea6-47ca-a2ec-4c3f42ab947e)

![Image](https://github.com/user-attachments/assets/d59d8fbd-6bdd-41ee-89a7-d16b7f6b3094)

Passing this back to Todo.

**Comment by davidclaveau** [comment]
@manuji this should be good for review! Looks like the issue was a small typo with the new exporter logic, and then an issue with how it request data with those start and end dates.

All the styling issues should be addressed now as well, so "Export all" should be a proper size, the calendar picker should be a little wider so months don't get clipped, and then some fixes to the date input box so it doesn't push items outside the box.

Let me know if it all looks good.

![Image](https://github.com/user-attachments/assets/9bcc24ec-1fba-46f1-b9f0-8337b4bbc9d4)
![Image](https://github.com/user-attachments/assets/02497259-f861-4956-8388-41cc7bbb27c8)

**Comment by manuji** [comment]
Tested on TEST: **Passed**
* verified that the issues have been fixed and the report can be exported. 
* Passing to PO review.

**Comment by LindsayMacfarlane** [comment]
Looks great! Thank you @meredithom @davidclaveau and @manuji !
